pkg/metacni/metacni.go

```diff
@@ -207,14 +210,54 @@ func popDevice(devicePool string, allocatedDevices map[string]*[]string)(string,
   return device, nil
 }
 
+func setIpv6Sysctls(args *cniArgs) error {
+  runtime.LockOSThread()
+  defer runtime.UnlockOSThread()
+  // save the current namespace
+  origNs, err := ns.GetCurrentNS()
+  if err != nil {
+    return errors.New("failed to get current network namespace due to:" + err.Error())
+  }
+  // enter to the Pod's network namespace
+  podNs, err := ns.GetNS(args.netns)
+  if err != nil {
+    return errors.New("failed to get Pod's network namespace due to:" + err.Error())
+  }
+  err = podNs.Set()
+  if err != nil {
+    return errors.New("failed to switch to Pod's network namespace due to:" + err.Error())
+  }
+  defer func() {
+    podNs.Close()
+    origNs.Set()
+  }()
+  // set sysctls
+  log.Println("IPv6 sysctl: net.ipv6.conf.all.disable_ipv6 = 0")
+  _, err = sysctl.Sysctl("net.ipv6.conf.all.disable_ipv6", "0")
```

Comment:
I would prefer per interface setting, rather than a generic enable it over all

Reply:
Done, but using invert logic.
We enable IPv6 on all interfaces (in order to run CNI smoothly), then we disable IPv6 where it is not needed.